linera-core/src/client/mod.rs

```diff
@@ -1979,11 +1977,18 @@ where
                 blobs,
             ))
         };
-        // Check the final block proposal. This will be cheaper after #1401.
-        self.client
-            .local_node
-            .handle_block_proposal(*proposal.clone())
-            .await?;
+        if manager
```

Comment:
In line 1970 we extract `manager.requested_locked` but we did that in line 1938 as well. Are these duplications necessary?